router.go

```diff
@@ -40,10 +41,12 @@ type defaultRouter struct {
 	Broker
 	Dealer
 
-	realms  map[URI]Realm
-	clients map[URI][]Session
-	closing bool
-	lastId  int
+	realmsLock  sync.RWMutex
```

Comment:
Since `RegisterRealm` isn't going to be called from multiple goroutines I don't think `realmsLock` is really required.